client/lib/pages/onboarding/onboarding_page.dart

```diff
@@ -128,10 +128,11 @@ class _OnboardingPageState extends State<OnboardingPage> {
   }
 
   Future<void> _onLegalDone(ContentStore store) async {
-    await UserPreferences().setTermsOfServiceCompleted(true);
+    await UserPreferences().setLegalAccepted();
     // Enable auto init so that analytics will work
     await FirebaseMessaging.instance.setAutoInitEnabled(true);
     if (!await UserPreferences().getOnboardingCompleted() ||
+        !await UserPreferences().getOnboardingCompletedV1() ||
```

Comment:
hmmm this logic may not be right when you add this.
consider when onboardingComplete=false, and onboardingCompetedV1=true, and analyticsEnabled=false
this will still analytics on, which is wrong

Reply:
Good catch. Fixed this logic:

1. Only enable analytics for first onboarding.... i.e. when both prior onboardings are false
1. Dropped `|| await UserPreferences().getAnalyticsEnabled()` in logic as enabling analytics when it's already enabled is completely redundant